docs/bsp.md

```diff
@@ -717,6 +736,10 @@ trait SourceItem {
     * directory belongs to this source item.
     */
   def uri: Uri
+  
+  /** Specifies whether uri denotes a directory or regular file. 
+    * If this is not set, and the uri does not point to an existing file, it should be treated as directory by default. */
+  def isDirectory: Option[Boolean]
```

Comment:
Are we sure we want to use `isDirectory` instead of `kind`?

Reply:
not feeling strongly, but as stated in #83 I think this is a case where we needn't expect expansion of types of files on this dimension :)